mmit/tree.py

```diff
@@ -250,8 +255,47 @@ def _optimal_split(node):
                                                                     self.margin, 0 if self.loss == "hinge" else 1)
 
                 # If that fails, something is wrong with the solver
-                if not np.allclose(left_costs[-1], right_costs[-1]) or not np.allclose(left_preds[-1], right_preds[-1]):
-                    raise RuntimeError("MMIT solver error. Please report this to the developers.")
+                if np.abs(left_costs[-1] - right_costs[-1]) > float_tol or np.abs(left_preds[-1] - right_preds[-1]) > float_tol:
```

Comment:
maybe add a comment about this being a runtime test? what are you checking? It is that the cost and prediction are the same at the end (when all data points are considered), right?